applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2464,4 +2464,38 @@ public static function allowedSortFields($Allowed = null) {
 
         return self::$AllowedSortFields;
     }
+
+
+    /**
+     * Tests whether a user has permission to view a specific discussion by checking category-specific permissions.
+     * Fires an event that can override the category view permission.
+     *
+     * @param object|integer $discussion The discussion ID or the discussion to test.
+     * @param string|integer $userId The ID of the user to test permission for. If empty, it defaults to Session user.
+     * @param string $categoryPermission The category permission to test against the user.
+     * @return bool Whether the user can view the discussion.
+     * @throws Exception
+     */
+    public function canViewDiscussion($discussion, $userId = '', $categoryPermission = 'Vanilla.Discussions.View') {
```

Comment:
The userID argument is used often here, and the category permission is rarely changed. If we change the order, then dev will have to call canView($discussion, 'Vanilla.Discussions.View', $userID); rather than simply canView($discussion, $userID);